// dlv-ng-services-menu/dlv-ng-services-menu/dlv-ng-services-menu.module.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CookiesService, MenuService, PermissionService, FilterByGroup, CenterService, AutoCompleteService } from './shared/services';
import { ClickOutsideDirective } from './services-menu/click-out-side.directive';
import { ServicesMenuComponent } from './services-menu/services-menu.component';
import { ServicesMenuOverlayComponent } from './services-menu-overlay/services-menu-overlay.component';
import { WaybillComponent } from './waybill-search/waybill-search.component';
import { CenterComponent } from './selected-center/selected-center.component';
import { CenterModalComponent } from './center-modal/center-modal.component';
import { AutoComplete } from './auto-complete-center/auto-complete.component';
import { ErrorComponent } from './error-alert/error-alert.component';
import { DynamicHeight } from './shared/directive';
export { MenuService, CenterService } from './shared/services';
var DlvNgServicesMenuModule = /** @class */ (function () {
    function DlvNgServicesMenuModule() {
    }
    DlvNgServicesMenuModule = __decorate([
        NgModule({
            imports: [
                CommonModule,
                FormsModule,
                ReactiveFormsModule
            ],
            declarations: [
                FilterByGroup,
                DynamicHeight,
                ServicesMenuComponent,
                ServicesMenuOverlayComponent,
                WaybillComponent,
                CenterComponent,
                CenterModalComponent,
                AutoComplete,
                ClickOutsideDirective,
                ErrorComponent
            ],
            exports: [
                DynamicHeight,
                ServicesMenuComponent,
                ServicesMenuOverlayComponent,
                WaybillComponent,
                CenterComponent,
                CenterModalComponent,
                AutoComplete,
                ClickOutsideDirective,
                ErrorComponent
            ],
            providers: [
                CookiesService,
                MenuService,
                PermissionService,
                CenterService,
                AutoCompleteService,
            ]
        })
    ], DlvNgServicesMenuModule);
    return DlvNgServicesMenuModule;
}());
export { DlvNgServicesMenuModule };
