// dlv-ng-auth/user-details/user-details.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { UserService, AuthService, AuthDetailsService } from '../shared/services';
import { ChangePasswordComponent } from '../change-password/change-password.component';
var DelhiveryUserDetailsComponent = /** @class */ (function () {
    function DelhiveryUserDetailsComponent(userService, authService, authDetailService) {
        this.userService = userService;
        this.authService = authService;
        this.authDetailService = authDetailService;
        this.onLogout = new EventEmitter();
        this.isChange = false;
    }
    DelhiveryUserDetailsComponent.prototype.ngOnInit = function () {
        var userDetails = this.authService.getUserProfile({ server: false });
        try {
            this.user = userDetails['first_name'] && userDetails['last_name']
                ? userDetails['first_name'] + " " + userDetails['last_name'] : this.getDisplayName(userDetails);
        }
        catch (err) {
            this.user = this.getDisplayName(userDetails);
        }
        this.shortName = userDetails['first_name'] && userDetails['last_name'] ? userDetails['first_name'][0] + userDetails['last_name'][0] : userDetails['first_name'][0] + userDetails['last_name'][1];
        this.avatarBackgroundcolor = this.userService.getBackgroundColor(this.user);
    };
    DelhiveryUserDetailsComponent.prototype.ngDoCheck = function () {
        var _this = this;
        this.authDetailService.getOn401()
            .subscribe(function () {
            _this.logout();
        });
    };
    DelhiveryUserDetailsComponent.prototype.logout = function () {
        var _this = this;
        this.authService.logout();
        setTimeout(function () {
            _this.onLogout.emit();
        }, 1);
    };
    DelhiveryUserDetailsComponent.prototype.getDisplayName = function (user) {
        var fullname = '';
        if (user['first_name'] && user['first_name'].length < 3) {
            fullname = user['first_name'] + " " + user['last_name'].substring(0, 10);
        }
        else if (user['first_name'] && user['first_name'].length >= 3) {
            fullname = user['first_name'];
        }
        else {
            fullname = user.username;
        }
        return fullname;
    };
    DelhiveryUserDetailsComponent.prototype.changePassword = function () {
        this.isChange = true;
    };
    DelhiveryUserDetailsComponent.prototype.callBack = function (data) {
        this.isChange = data;
    };
    __decorate([
        Output(),
        __metadata("design:type", Object)
    ], DelhiveryUserDetailsComponent.prototype, "onLogout", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], DelhiveryUserDetailsComponent.prototype, "isMaterialView", void 0);
    DelhiveryUserDetailsComponent = __decorate([
        Component({
            template: "\n    <div *ngIf=\"!isMaterialView\">\n        <div class=\"user-details-section invert\" *ngIf=\"!isChange\" [ngClass]=\"style\">\n        <div class=\"display-text\">\n          <i class=\"user-icon icon-dp hidden-xs\"></i>\n          <span class=\"user-name\">{{user}}</span>\n        </div>\n        <div class=\"logout\">\n          <span (click)=\"logout()\">Logout</span>\n        </div>\n      </div>\n    </div>\n\n    <!-- <change-password *ngIf=\"isChange\" [isChange]=\"isChange\" (changePasswordEmitter)=\"callBack($event)\"></change-password> -->\n\n\n    <div *ngIf=\"isMaterialView\" class=\"mat-profile\">\n       <mat-menu #profileMenu=\"matMenu\">\n        <!-- <button mat-menu-item disabled>Change password</button> -->\n        <button mat-menu-item (click)=\"logout()\">Logout</button>\n      </mat-menu>\n      <span class=\"mat-ava\" [ngStyle]=\"{'background': avatarBackgroundcolor || '#000'}\">{{shortName}}</span>\n      <button mat-flat-button [matMenuTriggerFor]=\"profileMenu\" class=\"profile-link\">\n        {{user}}&nbsp;<span class=\"material-icons\">expand_more</span></button>\n      <!-- <a [matMenuTriggerFor]=\"profileMenu\" class=\"profile-link\">\n        {{user}}&nbsp;<span class=\"material-icons\">expand_more</span>\n      </a> -->\n    </div>\n  ",
            styles: ["\n    .user-details-section{float:right;padding:0 20px 0 20px;height:40px;position:relative;border-left:2px solid transparent;z-index:2}.user-details-section .display-text{height:calc(100% - 16px);color:#515356;line-height:35px;float:left;margin:8px 0;position:relative;padding:0 10px 0 0}.user-details-section .display-text .user-icon{font-size:1.7em;margin-right:5px}.user-details-section .display-text:after{content:'';width:1px;height:calc(100% - 6px);background:#515356;position:absolute;top:3px;right:0}.user-details-section .logout{float:right;padding-left:10px;color:#515356;font-family:OpenSans-Semibold;margin:3px 0 2px;line-height:35px;cursor:pointer}.user-details-section .display-text .user-name{position:relative;top:-5px}.user-details-section.invert .display-text,.user-details-section.invert .display-text i,.user-details-section.invert .display-text span,.user-details-section.invert .logout,.user-details-section.invert .logout span{color:#ffffff}.user-details-section.invert .display-text:after{background:#ffffff}.mat-profile{float:right;margin-right:13px;background:#fff}a.profile-link{color:#696969}span.mat-ava{border-radius:50%;background:red;color:#fff;font-weight:600;font-size:15px;display:inline-block;height:38px;width:38px;line-height:38px;padding:0 10px;text-transform:uppercase}\n  "],
            selector: 'delhivery-user-details',
            entryComponents: [ChangePasswordComponent]
        }),
        __metadata("design:paramtypes", [UserService,
            AuthService,
            AuthDetailsService])
    ], DelhiveryUserDetailsComponent);
    return DelhiveryUserDetailsComponent;
}());
export { DelhiveryUserDetailsComponent };
