// dlv-ng-services-menu/dlv-material-services-menu/dlv-material-services-menu.module.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CookiesService, MatMenuService, PermissionService, FilterByGroup, MatCenterService, AutoCompleteService } from './shared/services';
import { ClickOutsideDirective } from './dlv-mat-services-menu/click-out-side.directive';
import { DlvMatServicesMenuComponent } from './dlv-mat-services-menu/services-menu.component';
import { DlvMatServicesMenuOverlayComponent } from './dlv-mat-services-menu-overlay/services-menu-overlay.component';
import { DlvMatWaybillComponent } from './dlv-mat-waybill-search/waybill-search.component';
import { DlvMatCenterComponent } from './dlv-mat-selected-center/selected-center.component';
import { DlvMatCenterModalComponent } from './dlv-mat-center-modal/center-modal.component';
import { DlvMatAutoCompleteComponent } from './dlv-mat-auto-complete-center/auto-complete.component';
import { DlvMatErrorComponent } from './dlv-mat-error-alert/error-alert.component';
import { DynamicHeight } from './shared/directive';
export { MatMenuService, MatCenterService } from './shared/services';
import { DLVMaterialModule } from 'dlv-material';
var DlvMaterialServicesMenuModule = /** @class */ (function () {
    function DlvMaterialServicesMenuModule() {
    }
    DlvMaterialServicesMenuModule = __decorate([
        NgModule({
            imports: [
                CommonModule,
                FormsModule,
                ReactiveFormsModule,
                DLVMaterialModule,
            ],
            declarations: [
                FilterByGroup,
                DynamicHeight,
                DlvMatServicesMenuComponent,
                DlvMatServicesMenuOverlayComponent,
                DlvMatWaybillComponent,
                DlvMatCenterComponent,
                DlvMatCenterModalComponent,
                DlvMatAutoCompleteComponent,
                ClickOutsideDirective,
                DlvMatErrorComponent,
            ],
            exports: [
                DynamicHeight,
                DlvMatServicesMenuComponent,
                DlvMatServicesMenuOverlayComponent,
                DlvMatWaybillComponent,
                DlvMatCenterComponent,
                DlvMatCenterModalComponent,
                DlvMatAutoCompleteComponent,
                ClickOutsideDirective,
                DlvMatErrorComponent
            ],
            providers: [
                CookiesService,
                MatMenuService,
                PermissionService,
                MatCenterService,
                AutoCompleteService,
            ],
            entryComponents: [DlvMatCenterModalComponent]
        })
    ], DlvMaterialServicesMenuModule);
    return DlvMaterialServicesMenuModule;
}());
export { DlvMaterialServicesMenuModule };
