// dlv-ng-services-menu/dlv-ng-services-menu/waybill-search/waybill-search.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input } from '@angular/core';
import { FormGroup, FormControl } from '@angular/forms';
import { MenuService } from '../shared/services';
var WaybillComponent = /** @class */ (function () {
    // pattern = "([A-Z]{1,3}[0-9]{10,12})(\s*,\s*([A-Z]{1,3}\d{10,12}))*$"
    function WaybillComponent(menuService) {
        this.menuService = menuService;
        this.waybills = [];
    }
    WaybillComponent.prototype.ngOnInit = function () {
        this.createFormControls();
        this.createForm();
    };
    /**
     * @name createFormControls
     * @description This method is assigning value to input field
     */
    WaybillComponent.prototype.createFormControls = function () {
        this.waybillNo = new FormControl('', []);
    };
    /**
     * @name createForm
     * @description It is containing waybill no
     */
    WaybillComponent.prototype.createForm = function () {
        this.waybillForm = new FormGroup({
            waybillNo: this.waybillNo,
        });
    };
    /**
     * @name searchWaybill
     * @description this method is using to get the searched waybill detail on seperate window.
     */
    WaybillComponent.prototype.searchWaybill = function () {
        var waybill = this.waybillForm.value;
        var urlEndPoint = this.menuService.getUrlLink()[this.menuService.getOptions().environment];
        window.open(urlEndPoint + "p/list/1?q=" + waybill.waybillNo);
    };
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], WaybillComponent.prototype, "placeHolder", void 0);
    WaybillComponent = __decorate([
        Component({
            selector: 'delhivery-waybill-search',
            template: "\n    <div class=\"waybill\">\n        <form [formGroup]=\"waybillForm\">\n\n\n            <div class=\"form-search\">\n                <input type=\"text\" formControlName=\"waybillNo\" (keyup.enter)=\"searchWaybill()\" placeholder=\"{{placeHolder}}\" name=\"waybillNo\" />\n                <span (click)=\"searchWaybill()\" class=\"fa fa-search\"></span>\n            </div>\n\n        </form>\n    </div>\n  ",
            styles: ["\n    .waybill{position:relative;margin:6px 0 0 0}.waybill .form-search{position:relative}.waybill .form-search input{box-shadow:none !important;border-radius:0;border:1px solid #fff;font-size:13px;line-height:21px;color:#000;background:#fff;padding:3px 22px 3px 8px;width:100%;height:28px;box-sizing:border-box}.waybill .form-search input:focus{border:1px solid #fff;outline:none}.waybill .form-search span{position:absolute;right:0;top:0;color:#808080;padding:6px;font-size:15px;cursor:pointer}.waybill .form-search span:hover{color:#000}.waybill .form-search span.has-error{color:#ff0000;pointer-events:none}.waybill .errMsg{margin:4px 0;color:red;font-size:12px}\n  "]
        }),
        __metadata("design:paramtypes", [MenuService])
    ], WaybillComponent);
    return WaybillComponent;
}());
export { WaybillComponent };
