// dlv-ng-auth/shared/services/auth.service.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Injectable } from '@angular/core';
// Custom Services import
import { AuthDetailsService } from '../services/auth-details.service';
import { UserService } from '../services/user.service';
var AuthService = /** @class */ (function () {
    function AuthService(authDetailsService, userService) {
        this.authDetailsService = authDetailsService;
        this.userService = userService;
    }
    AuthService.prototype.init = function (customOptions) {
        this.authDetailsService.setAuthConstants(customOptions);
    };
    ;
    AuthService.prototype.isAuthenticated = function () {
        return this.authDetailsService.isAuthenticated();
    };
    ;
    // Get profile form server or JWT decode
    AuthService.prototype.getUserProfile = function (data) {
        if (data && data.hasOwnProperty('server') && data.server) {
            return this.userService.getUserFromServer();
        }
        else {
            return this.userService.getUserFromJwt();
        }
    };
    ;
    // Change Password
    AuthService.prototype.changePassword = function (data) {
        return this.userService.changePassword(data);
    };
    ;
    // Logout function
    AuthService.prototype.logout = function () {
        this.authDetailsService.deleteJWT();
    };
    ;
    AuthService = __decorate([
        Injectable(),
        __metadata("design:paramtypes", [AuthDetailsService, UserService])
    ], AuthService);
    return AuthService;
}());
export { AuthService };
