// dlv-material/esm5/lib/label/label.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input, ChangeDetectionStrategy, } from '@angular/core';
var MatLabel = /** @class */ (function () {
    function MatLabel() {
    }
    /**
     * @return {?}
     */
    MatLabel.prototype.closeHandler = /**
     * @return {?}
     */
    function () { };
    MatLabel.decorators = [
        { type: Component, args: [{
                    selector: 'mat-label',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    template: "\n  <span [class]=\"'label label-' + type \">{{label}} \n  </span>",
                    styles: [".label{margin-left:5px}.label-success{background-color:#0fb687}.label-info{background-color:#3aafd1}.label-warning{background-color:#f2b50e}.label-danger{background-color:#fa5c7c}.label-success-light{background-color:#fff;border:1px solid #0fb687;color:#0fb687;padding:2px 8px}.label-warning-light{background-color:#fff;border:1px solid #f2b50e;color:#f2b50e;padding:2px 8px}.label-info-light{background-color:#fff;border:1px solid #3aafd1;color:#3aafd1;padding:2px 8px}.label-danger-light{background-color:#fff;border:1px solid #fa5c7c;color:#fa5c7c;padding:2px 8px}"]
                }] }
    ];
    /** @nocollapse */
    MatLabel.ctorParameters = function () { return []; };
    MatLabel.propDecorators = {
        type: [{ type: Input }],
        label: [{ type: Input }]
    };
    return MatLabel;
}());
export { MatLabel };
if (false) {
    /** @type {?} */
    MatLabel.prototype.type;
    /** @type {?} */
    MatLabel.prototype.label;
}
