// dlv-ng-services-menu/dlv-material-services-menu/shared/services/constant.js
export var Constant = function () {
    return Object.freeze({
        UMS_API: {
            'local': 'https://api-stage-ums.delhivery.com',
            'dev': 'https://api-stage-ums.delhivery.com',
            'prod': 'https://api-ums.delhivery.com',
            'staging': 'https://api-stage-ums.delhivery.com',
            'qa': 'https://api-stage-ums.delhivery.com'
        },
        HQ_API: {
            'local': 'https://staging-express.delhivery.com',
            'dev': 'https://staging-express.delhivery.com',
            'prod': 'https://hq.delhivery.com',
            'staging': 'http://staging-express.delhivery.com',
            'qa': 'http://staging-express.delhivery.com'
        },
        PERFORMANCE_API: {
            'local': 'https://scores-dev.delhivery.com/',
            'dev': 'https://scores-dev.delhivery.com/',
            'prod': 'https://scores.delhivery.com/',
            'staging': 'https://scores-dev.delhivery.com/',
            'qa': 'https://scores-dev.delhivery.com/',
        },
        FAAS_API: {
            'local': 'https://faas-dev-api.delhivery.com/',
            'dev': 'https://faas-dev-api.delhivery.com/',
            'prod': 'https://faas-api.delhivery.com/',
            'staging': 'https://faas-dev-api.delhivery.com/',
            'qa': 'https://faas-dev-api.delhivery.com/'
        },
        LINK_ADDRESS: {
            'local': '/',
            'dev': 'https://staging-express.delhivery.com/',
            'prod': 'https://hq.delhivery.com/',
            'staging': 'http://staging-express.delhivery.com/',
            'qa': 'http://qa2-express.delhivery.com/',
        },
        CONFIG_API: {
            'local': 'https://api-dev-console.delhivery.com',
            'dev': 'https://api-dev-console.delhivery.com',
            'prod': 'https://api-console.delhivery.com',
            'staging': 'https://api-dev-console.delhivery.com',
            'qa': 'https://api-dev-console.delhivery.com'
        }
    });
};
